CRM/Member/Form/MembershipView.php

```diff
@@ -344,6 +344,11 @@ public function preProcess() {
         $displayName .= ' (' . ts('default organization') . ')';
       }
 
+      // Add recurring contribution
+      if (!empty($values['contribution_recur_id'])) {
+        $this->assign('recurContribution', $values['contribution_recur_id']);
+      }
```

Comment:
@mattwire I think you don't need to assign it to separate smarty variable, as below at [L406](https://github.com/civicrm/civicrm-core/pull/11365/files#diff-32ac549e1df6c8bda04ab496a759dd63R406) the whole $values array assigned to smarty. So directly call  
```patch
 +     {if $contribution_recur_id}
 +          <tr>
 +            <td class="label">{ts}Recurring Contribution{/ts}</td>
 +            <td>
 +              <a class="crm-hover-button" href='{crmURL p="civicrm/contact/view/contributionrecur" q="reset=1&id=`$recurContribution`&cid=`$contactId`&context=contribution"}'>View Recurring Contribution</a>
 +            </td>
 +          </tr>
 +        {/if}
```
should work. What do you think?